File: payloadGenerator.js
/**
 * node payloadGenerator.js path/to/privateKeys.json numberOfTransactions MessageWordCount
 */
console.time();
const nem = require("nem2-sdk");
const jsJoda = require("js-joda");
const Account = nem.Account,
  Deadline = nem.Deadline,
  NetworkType = nem.NetworkType,
  PlainMessage = nem.PlainMessage,
  TransferTransaction = nem.TransferTransaction,
  XEM = nem.XEM
;

const randInt = (max) => Math.floor(Math.random() * Math.floor(max));
const createString = (len) => "A".repeat(len);
const payloadCreater = () => {
  const deadline = Deadline.create(1439, jsJoda.ChronoUnit.MINUTES);
  return (account, recipient, amount, message) => {
    const xem = [XEM.createRelative(amount)];
    const msg = PlainMessage.create(message);
    const transferTransaction = TransferTransaction.create(
      deadline,
      recipient,
      xem,
      msg,
      NetworkType.MIJIN_TEST,
    );
    const signedTransaction = account.sign(transferTransaction);
    return signedTransaction.payload;
  }
}
const payloadCreaterWithFixed = (amount, message) => {
  const deadline = Deadline.create(1439, jsJoda.ChronoUnit.MINUTES);
  const xem = [XEM.createRelative(amount)];
  const msg = PlainMessage.create(message);
  return (account, recipient) => {
    const transferTransaction = TransferTransaction.create(
      deadline,
      recipient,
      xem,
      msg,
      NetworkType.MIJIN_TEST,
    );
    const signedTransaction = account.sign(transferTransaction);
    return signedTransaction.payload;
  }
}
const accountsPicker = (keys) => {
  const accounts = keys.map(k => Account.createFromPrivateKey(k, NetworkType.MIJIN_TEST));
  return (num) => [accounts[randInt(keys.length)], accounts[randInt(keys.length)]]
};

const privateKeys = require(process.argv[2] || "./privateKeys.json");
const num = process.argv[3] || 10;
const len = process.argv[4] || 0; // 1023;
const createPayload = payloadCreater();
const createPayloadWithFixed = payloadCreaterWithFixed(
  randInt(1),
  createString(randInt(len))
);
const pick2Accounts = accountsPicker(privateKeys);

const [sender, _] = pick2Accounts();
console.error(sender.address.plain())
for (let i = 0; i < num; i++) {
  const [_, recipient] = pick2Accounts();
  // const payload = createPayload(sender, recipient.address, randInt(5), createString(len));
  const payload = createPayloadWithFixed(sender, recipient.address);
  console.log(payload);
}

File: payloadGenerator.go
package main

import (
	"encoding/json"
	"flag"
	"fmt"
	"github.com/proximax-storage/nem2-sdk-go/sdk"
	"io/ioutil"
	"math/rand"
	"os"
	"time"
)

func main() {
	var path = flag.String("f", "./privateKeys.json", "Collection of PrivateKeys")
	var num = flag.Int("n", 10, "number of transactions")
	var msg = flag.String("m", "", "message string")
	flag.Parse()
	bytes, err := ioutil.ReadFile(*path)
	if err != nil {
		fmt.Println(err.Error())
		os.Exit(1)
	}
	var keys []string
	json.Unmarshal(bytes, &keys)

	networkType := sdk.MijinTest
	accounts := make([]sdk.Account, len(keys), len(keys))
	for idx, key := range keys {
		acc, err := sdk.NewAccountFromPrivateKey(key, networkType)
		if err != nil {
			panic(err)
		}
		accounts[idx] = *acc
	}

	//sender := accounts[rand.Intn(len(accounts))]
	sender := accounts[1]
	fmt.Fprint(os.Stderr, sender.Address.Pretty())
	//fmt.Println(sender.Address.Pretty())
	for i := 0; i < *num; i++ {
		// sender := accounts[rand.Intn(len(accounts))]
		recipient := accounts[rand.Intn(len(accounts))]
		ttx, err := sdk.NewTransferTransaction(
			sdk.NewDeadline(time.Hour*23+time.Minute*59),
			sdk.NewAddress(recipient.Address.Pretty(), networkType),
			[]*sdk.Mosaic{sdk.Xem(0)},
			sdk.NewPlainMessage(*msg),
			networkType,
		)
		stx, err := sender.Sign(ttx)
		if err != nil {
			panic(fmt.Errorf("TransaferTransaction signing returned error: %s", err))
		}
		fmt.Println(stx.Payload)
	}
}
